src/figuresClasses.ts

```diff
@@ -1,19 +1,78 @@
 export interface Figure {
-
+  shape: string,
+  color: string,
+  getArea(): number,
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: string = 'triangle';
+
+  constructor(
+    public color: string,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Invalid Number');
+    }
+
+    if (a + b <= c || b + c <= a || a + c <= b) {
+      throw new Error('sides 1, 2 and 3 can`t form a triangle');
+    }
+  }
+
+  getArea(): number {
+    const per: number = (this.a + this.b + this.c) / 2;
```

Comment:
I think if you do the destructuring for the object, your code will be readable
```suggestion
   const { a, b, c } = this;
    const per: number = (a + b + c) / 2;
```